lib/active_merchant/billing/gateways/mundipagg.rb

```diff
@@ -155,7 +155,7 @@ def add_payment(post, payment, options)
         post[:customer][:name] = payment.name if post[:customer]
         post[:customer_id] = parse_auth(payment)[0] if payment.is_a?(String)
         post[:payment] = {}
-        post[:payment][:gateway_affiliation_id] = @options[:gateway_id] if @options[:gateway_id]
+        post[:payment][:gateway_affiliation_id] = options[:gateway_affiliation_id] if options[:gateway_affiliation_id]
```

Comment:
Would there be any value to falling back to the GW option if the txn-level option is omitted (esp. if there's a chance of impacting other users of AM who might benefit from a backwards-compatible fallback)?

Reply:
That's a good point. Since it was never required on initialize and we aren't messing with that signature anyway, there's no harm in leaving a naive fallback as you suggest.

Reply:
Updated with fallback